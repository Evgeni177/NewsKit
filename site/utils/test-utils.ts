import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
