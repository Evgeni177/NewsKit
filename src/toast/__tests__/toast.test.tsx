import React from 'react';
import {fireEvent, waitForElementToBeRemoved} from '@testing-library/react';
import hotToast from 'react-hot-toast';
import {ToastProvider, ToastProviderProps, ToastPosition, toast} from '..';
import {renderWithTheme} from '../../test/test-utils';

const ToastExample = ({
  action = () => null,
  providerProps,
}: {
  action?: Function;
  providerProps?: ToastProviderProps;
}) => (
  <React.Fragment>
    <button type="button" data-testid="action" onClick={() => action()}>
      action
    </button>

    <ToastProvider {...providerProps} />
  </React.Fragment>
);

describe('Toast', () => {
  describe('Component', () => {
    // TODO in other PR
  });

  describe('API', () => {
    afterEach(() => {
      hotToast.dismiss();
    });

    const positions: ToastPosition[] = [
      'top-right',
      'top-center',
      'top-left',
      'bottom-right',
      'bottom-center',
      'bottom-left',
    ];

    positions.forEach(position => {
      const action = () => toast(<div>message</div>);
      test(`render toast at ${position} position`, () => {
        const {asFragment, getByTestId, unmount} = renderWithTheme(
          ToastExample,
          {
            providerProps: {
              position,
              horizontalOffset: '10px',
              verticalOffset: '20px',
              autoHideDuration: 4000,
            },
            action,
          },
        );
        const actionBtn = getByTestId('action');
        fireEvent.click(actionBtn);
        expect(asFragment()).toMatchSnapshot();
        unmount();
      });
    });

    test('toast adds component to DOM', () => {
      const messageText = 'test message';
      const action = () => toast(<div>{messageText}</div>);
      const {getByTestId, getByText} = renderWithTheme(ToastExample, {action});

      const actionBtn = getByTestId('action');
      fireEvent.click(actionBtn);
      const toastMessage = getByText(messageText);
      expect(toastMessage).toBeDefined();
    });

    test('toast can be removed from DOM via button', async () => {
      const messageText = 'test message with button';
      const action = () =>
        toast(
          ({onClose}) => (
            <div>
              {messageText}
              <button
                type="button"
                data-testid="close"
                onClick={() => onClose()}
              >
                close
              </button>
            </div>
          ),
          {autoHideDuration: 1500},
        );
      const {getByTestId, getByText} = renderWithTheme(ToastExample, {
        action,
      });

      const actionBtn = getByTestId('action');
      fireEvent.click(actionBtn);
      const closeBtn = getByTestId('close');
      expect(closeBtn).toBeDefined();
      fireEvent.click(closeBtn);

      await waitForElementToBeRemoved(() => getByText(messageText));
    });
  });
});
