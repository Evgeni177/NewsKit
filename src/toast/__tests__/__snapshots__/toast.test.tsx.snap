// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast API render toast at bottom-center position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  bottom: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

.emotion-2 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-2>* {
  pointer-events: none;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast API render toast at bottom-left position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  bottom: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

.emotion-2 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-2>* {
  pointer-events: none;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast API render toast at bottom-right position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, 100%, 0) scale(0.1);
    -moz-transform: translate3d(0, 100%, 0) scale(0.1);
    -ms-transform: translate3d(0, 100%, 0) scale(0.1);
    transform: translate3d(0, 100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,100%,0) scale(0.1);
    -moz-transform: translate3d(0,100%,0) scale(0.1);
    -ms-transform: translate3d(0,100%,0) scale(0.1);
    transform: translate3d(0,100%,0) scale(0.1);
    opacity: 0;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  bottom: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

.emotion-2 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 8px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  bottom: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-2>* {
  pointer-events: none;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="bottom: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast API render toast at top-center position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,-100%,0) scale(0.1);
    -moz-transform: translate3d(0,-100%,0) scale(0.1);
    -ms-transform: translate3d(0,-100%,0) scale(0.1);
    transform: translate3d(0,-100%,0) scale(0.1);
    opacity: 0;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  top: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

.emotion-2 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-2>* {
  pointer-events: none;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast API render toast at top-left position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,-100%,0) scale(0.1);
    -moz-transform: translate3d(0,-100%,0) scale(0.1);
    -ms-transform: translate3d(0,-100%,0) scale(0.1);
    transform: translate3d(0,-100%,0) scale(0.1);
    opacity: 0;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }

  100% {
    -webkit-transform: translate3d(0,-100%,0) scale(0.1);
    -moz-transform: translate3d(0,-100%,0) scale(0.1);
    -ms-transform: translate3d(0,-100%,0) scale(0.1);
    transform: translate3d(0,-100%,0) scale(0.1);
    opacity: 0;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  top: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 8px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  top: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

.emotion-2 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 8px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  top: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-2>* {
  pointer-events: none;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
    <div
      class="emotion-2"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast API render toast at top-right position 1`] = `
<DocumentFragment>
  <button
    data-testid="action"
    type="button"
  >
    action
  </button>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate3d(0, -100%, 0) scale(0.1);
    -moz-transform: translate3d(0, -100%, 0) scale(0.1);
    -ms-transform: translate3d(0, -100%, 0) scale(0.1);
    transform: translate3d(0, -100%, 0) scale(0.1);
    opacity: 0;
  }

  100% {
    -webkit-transform: translate3d(0, 0, 0) scale(1);
    -moz-transform: translate3d(0, 0, 0) scale(1);
    -ms-transform: translate3d(0, 0, 0) scale(1);
    transform: translate3d(0, 0, 0) scale(1);
    opacity: 1;
  }
}

.emotion-0 {
  position: fixed;
  pointer-events: none;
  left: 10px;
  right: 10px;
  top: 20px;
}

.emotion-1 {
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 8px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  top: 0;
  transition-property: top,bottom;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(.5, 0, .5, 1);
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
  -webkit-animation-timing-function: cubic-bezier(.5, 0, .5, 1);
  animation-timing-function: cubic-bezier(.5, 0, .5, 1);
}

.emotion-1>* {
  pointer-events: initial;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      style="top: 0px;"
    >
      <div>
        message
      </div>
    </div>
  </div>
</DocumentFragment>
`;
